packages/material-ui/src/test-utils/createMount.js

```diff
@@ -31,8 +31,21 @@ class Mode extends React.Component {
   }
 }
 
+let warnedOnce = false;
+
 // Generate an enhanced mount function.
 export default function createMount(options = {}) {
+  if (!warnedOnce && process.env.INTERNAL !== 'true') {
```

Comment:
Do we even use these internally? I thought I completely forked them in `test/utils`. Could you follow the breaking approach instead so that we don't pollute the environment with more variables?

Reply:
It seems to depend on the tests. I could find:

https://github.com/mui-org/material-ui/blob/7b9c692a2d9d6ecef8831adee6e1f9660070ede5/packages/material-ui/src/AccordionDetails/AccordionDetails.test.js#L3

> Could you follow the breaking approach

Could you expand on this approach? I don't follow what you are referring to.

Reply:
For deprecations the simplest approach is to cherry-pick the breaking change and then revert all the breaking changes to the public API. In this case:

1. cherry-pick https://github.com/mui-org/material-ui/pull/21855
2. Add a deprecation warning to methods in `'@material-ui/core/test-utils'` instead of removing them

Conceptually we want to move from `master` to `next` and it is easier for now and for feature integration of `master` into `next` to re-use as much as possible from the commit that made `master` and `next` diverge. The goal is to move `master` and `next` "closer" together. Your approach is moving them farther apart making it harder to maintain both branches.

Reply:
I see, thanks for the explanation. You are right, optimizing for the gap between `master` and `next` was definitely not something I took into account in the first attempt. I was going for these two objectives:

- keeping the `master` and `v4-deprecations` branches as close as possible so we can merge `v4-deprecations` back into `master` easily.
- spend as little as time as I could get away with it (aiming for the good enough, or almost good enough)

Reply:
I have looked at the path proposed, I would need to resolve these conflicts:

<img width="516" alt="Capture d’écran 2020-12-21 à 21 43 23" src="https://user-images.githubusercontent.com/3165635/102820481-9c0b1880-43d5-11eb-908f-70496942ed8a.png">

I have a proposal: If temporarily not having an extra variable in the test env important for you, we can close this PR. You resume the effort later on with the alternative you proposed https://github.com/mui-org/material-ui/pull/24074#discussion_r546674073. Otherwise, we can move forward as is, to save time. I personally don't want to spend time resolving 74 conflicts for the outcome we could get in exchange. What do you think? This way you can control the quality of the outcome, nor do we need to rush it :).